kernel/src/scan/mod.rs

```diff
@@ -681,73 +682,6 @@ pub fn selection_vector(
     Ok(deletion_treemap_to_bools(dv_treemap))
 }
 
-/// Transform the raw data read from parquet into the correct logical form, based on the provided
-/// global scan state and partition values
-pub fn transform_to_logical(
-    engine: &dyn Engine,
-    data: Box<dyn EngineData>,
-    global_state: &GlobalScanState,
-    partition_values: &HashMap<String, String>,
-) -> DeltaResult<Box<dyn EngineData>> {
-    let state_info = get_state_info(
-        &global_state.logical_schema,
-        &global_state.partition_columns,
-    )?;
-    transform_to_logical_internal(
-        engine,
-        data,
-        global_state,
-        partition_values,
-        &state_info.all_fields,
-        state_info.have_partition_cols,
-    )
-}
-
-// We have this function because `execute` can save `all_fields` and `have_partition_cols` in the
-// scan, and then reuse them for each batch transform
-fn transform_to_logical_internal(
-    engine: &dyn Engine,
-    data: Box<dyn EngineData>,
-    global_state: &GlobalScanState,
-    partition_values: &std::collections::HashMap<String, String>,
-    all_fields: &[ColumnType],
-    have_partition_cols: bool,
-) -> DeltaResult<Box<dyn EngineData>> {
-    let physical_schema = global_state.physical_schema.clone();
-    if !have_partition_cols && global_state.column_mapping_mode == ColumnMappingMode::None {
```

Comment:
IIRC, this code was annoying when I added column mapping support for expression eval. I _think_ this was the only code left that specifically tracked or cared about column mapping mode, because of the new way logical -> physical transforms are performed. 

Recommend to audit the caller chain and see what other code can be simplified, now that we don't need column mapping logic here any more.